orchestra/migrations/0080_completed_project_status.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.12 on 2018-08-14 17:59
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+PROJECT_STATUS_COMPLETED = 4
+PROJECT_STATUS_ABORTED = 2
+
+TASK_STATUS_COMPLETED = 5
+
+def mark_completed_projects(apps, schema_editor):
+    Project = apps.get_model('orchestra', 'Project')
+    Task = apps.get_model('orchestra', 'Task')
+    for project in Project.objects.all():
+        workflow_version = project.workflow_version
+        all_steps = workflow_version.steps.all()
+        completed_tasks = Task.objects.filter(
+            status=TASK_STATUS_COMPLETED,
+            project=project)
+        completed_step_slugs = set(
+            completed_tasks.values_list('step__slug', flat=True))
+
+        if (len(all_steps) == len(completed_step_slugs) and
+                project.status != PROJECT_STATUS_ABORTED):
```

Comment:
isn't it all ACTIVE projects?

Reply:
This will include all "completed" ACTIVE and PAUSED projects. Do you mean I should just filter ACTIVE projects (which makes sense too because PAUSED projects shouldn't be completed anyway)?